src/node_file.h

```diff
@@ -79,12 +86,30 @@ class BindingData : public SnapshotableObject {
   SERIALIZABLE_OBJECT_METHODS()
   SET_BINDING_ID(fs_binding_data)
 
+  static void LegacyMainResolve(
+      const v8::FunctionCallbackInfo<v8::Value>& args);
+
+  static void CreatePerIsolateProperties(IsolateData* isolate_data,
+                                         v8::Local<v8::ObjectTemplate> ctor);
+  static void RegisterExternalReferences(ExternalReferenceRegistry* registry);
+
   void MemoryInfo(MemoryTracker* tracker) const override;
   SET_SELF_SIZE(BindingData)
   SET_MEMORY_INFO_NAME(BindingData)
 
  private:
   InternalFieldInfo* internal_field_info_ = nullptr;
+
+  static FilePathIsFileReturnType FilePathIsFile(Environment* env,
+                                                 const std::string& file_path);
+
+  static const std::array<std::string, 10> legacy_main_extensions;
```

Comment:
can you move the std::array contents to here, instead of defining them in .cc?

Reply:
Do not compile:

```
../../src/node_file.h:110:44: error: in-class initialization of static data member ‘const std::array<std::__cxx11::basic_string<char>, 10> node::fs::BindingData::legacy_main_extensions’ of non-literal type
  110 |   static const std::array<std::string, 10> legacy_main_extensions = {
      |                                            ^~~~~~~~~~~~~~~~~~~~~~
../../src/node_file.h:120:12: error: temporary of non-literal type ‘std::__cxx11::basic_string<char>’ in a constant expression
  120 |     ".node"};
```

Maybe it could compile if we were using `constexpr`, but since it didn't work with `std::string` :/

Reply:
This article can help: https://stackoverflow.com/questions/71882752/c-constexpr-stdarray-of-string-literals

Reply:
Yoo could use std::string_view instead of string as well.

Reply:
I use this array of strings to concat with the file path, it should be string since using string_view will lead me to do `std::string(legacy_main_extensions[i])`, which I don't think is a good behavior.

What do you think?

Reply:
yeah you are right. lets at least move the declaration to header file.

Reply:
To be able to do that, I need to move the declaration of `legacy_main_extensions` outside of `BindingData`, it's okay?